File: chinese-auction-web/src/app/models/winner.ts
import { Gift } from './gift';
import { User } from './user';

export interface Winner {
    id: number;
    giftId: number;
    gift: Gift;
    userId: number;
    user: User;
}

//פרטי המשתמש הזוכה למתנה ספיצפית - התאימה עם WinnerResultDto


export interface WinnerByGift {
    giftId: number;
    giftName: string;
    winnerUserId: number;        // ✅ אם לא int
    winnerName: string;          // ✅ string
    winnerEmail: string;         // ✅ string
    totalTickets: number;        // ✅ number בלבד
    drawDate: Date;              // ✅ Date
}

File: chinese-auction-api/ChineseAuction.Api/Models/Winner.cs
namespace ChineseAuction.Api.Models
{
    public class Winner
    {
        public int Id { get; set; } // ID זוכה

        public int GiftId { get; set; } // קשר למתנה: איזו מתנה זכה
        public Gift Gift { get; set; } = null!; // קשר למתנה: איזו מתנה זכה

        public int UserId { get; set; } // קשר לרוכש: מי הזוכה במתנה
        public User User { get; set; } = null!;// קשר לרוכש: מי הזוכה במתנה
    }
}
